docs: format the deprecation section in DOM cheatsheet (#846)
testing-library/testing-library-docs: docs/dom-testing-library/cheatsheet.mdx

> - **wait** (Promise) retry the function within until it stops throwing or
>   times
> - **waitForElement** (Promise) retry the function until it returns an element

> >   or an array of elements. `findBy` and `findAllBy` queries are async and
> >   retry until either a timeout or if the query returns successfully; they wrap
> >   `waitForElement`

Reviewer: I think this reads better?

```suggestion
>   or an array of elements. The `findBy` and `findAllBy` queries are async and
>   retry until the query returns successfully, or when the query times out; they wrap
>   `waitForElement`
```

Author: Yes it does. Thank you! 🙂
